Utils/HandlerUtil.py

```diff
@@ -201,24 +201,19 @@ def _parse_config(self, ctxt):
                 cms_cmd = 'openssl cms -inform DER -decrypt -recip {0} -inkey {1}'.format(cert,pkey)
                 smime_cmd = 'openssl smime -inform DER -decrypt -recip {0} -inkey {1}'.format(cert,pkey)
 
-                protected_settings_str = None
+                protected_settings_str = ''
                 for decrypt_cmd in [cms_cmd, smime_cmd]:
                     try:
-                        session = subprocess.Popen([decrypt_cmd], shell=True,
-                                                stdin=subprocess.PIPE,
-                                                stderr=subprocess.STDOUT,
-                                                stdout=subprocess.PIPE)
-                        output = session.communicate(unencodedSettings)
-                        # success only if return code is 0 and we have output
-                        if session.returncode == 0 and output[0]:
-                            protected_settings_str = output[0]
+                        output = waagent.RunSendStdin(decrypt_cmd, unencodedSettings)
+                        if output[0] == 0 and output[1]:
+                            protected_settings_str = output[1]
                             if decrypt_cmd == cms_cmd:
                                 self.log('Decrypted protectedSettings using openssl cms.')
                             else:
                                 self.log('Decrypted protectedSettings using openssl smime fallback.')
                             break
                         else:
-                            self.log('Attempt to decrypt protectedSettings with "{0}" failed (rc={1}).'.format(decrypt_cmd, session.returncode))
+                            self.log('Attempt to decrypt protectedSettings with "{0}" failed (rc={1}).'.format(decrypt_cmd, output[0]))
```

Comment:
this will be hit even when output[0] is null,and when it is 0 since you are checking on output[1] against null , should not it also be checked? what happens in those cases? i think it would still be exception that you want to catch/avoid